### PR TITLE
Issue #142 Introduce oc-env command

### PR DESCRIPTION
**Pull request by hferentschik**
Addresses issue #142

### CODE REVIEW

**Comment by gbraad** [comment]
I noticed side of the boilerplate show a copyright of 2016. I will still merge as this is not a breaking issue...

**Comment by gbraad** [comment]
Not merging: test for ocpath fails

**Comment by hferentschik** [comment]
> Not merging: test for ocpath fails

Right, on Windows this one cannot run. Give me a minute.

**Comment by hferentschik** [comment]
> I noticed side of the boilerplate show a copyright of 2016. I will still merge as this is not a breaking issue...

Let me fix that as well then

**Comment by gbraad** [comment]
centos CI fails the scenario? Can't go into more details now.

**Comment by hferentschik** [comment]
> centos CI fails the scenario? Can't go into more details now.

That's on and off at the moment. We thought we already addressed it by retrying the command. I can increase the retry count and we can see whether we get less failures. We went from failing every time, to every third time or so. However, we should take a closer look at this test again, since I am a bit suspicious why a retry and wait is required.

**Comment by hferentschik** [comment]
retest this please

**Comment by hferentschik** [comment]
@gbraad I actually disabled the failing integration test scenario and created https://github.com/minishift/minishift/issues/837 to follow up on this. It fails too often. Either the implementation for retrying has a bug or there is something else going on which is not addressed by the retry.

File: cmd/minishift/cmd/oc_env_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// +build !windows
```
Comment (budhrg) [inline cmd/minishift/cmd/oc_env_test.go]: Looks like this test is not passing Windows. What is the issue there?

File: cmd/minishift/cmd/oc_env_test_windows.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/*
```
Comment (budhrg) [inline cmd/minishift/cmd/oc_env_test_windows.go]: :+1:  Works

File: cmd/minishift/cmd/oc_env_test.go (hunk 1 above)
Comment (hferentschik) [inline cmd/minishift/cmd/oc_env_test.go]: It cannot pass on windows. It need to be excluded

**Comment by gbraad** [comment]
OK. LGTM.
The test has passed by several times I noticed while glancing at IRC during my leave. Needs a real fix. Agree to disable.

**Comment by hferentschik** [comment]
@gbraad green light :-)

File: docs/source/using/interacting-with-openshift.adoc [hunk 3]
```diff
@@ -24,14 +24,15 @@ For details about managing the Minishift VM, see the link:../using/managing-mini
 The `minishift start` command creates an OpenShift instance using the
 https://github.com/openshift/origin/blob/master/docs/cluster_up_down.md[cluster up] approach.
 
-For this purpose it copies the `oc` binary onto your host. You will find it under
-`~/.minishift/cache/oc/{openshift-version}/oc` provided you
-use the default version of OpenShift. You can add this
-binary to your `PATH` variable in order to use `oc`, for example:
+For this purpose it copies the `oc` binary onto your host. You will find it under `~/.minishift/cache/oc/{openshift-version}/oc`, provided you use Minishift's default version of OpenShift.
+You can add this binary to your `PATH` using link:../command-ref/minishift_oc-env{outfilesuffix}[`minishift oc-env`] which displays the command you need to type into your shell.
+The output of `oc_env` will differ depending on OS and shell type.
```
Comment (budhrg) [inline docs/source/using/interacting-with-openshift.adoc]: here it should be `oc-env`   not `oc_env`.

**Comment by gbraad** [comment]
Thanks

**Pull request merged**